cirq/circuits/circuit.py

```diff
@@ -211,7 +213,9 @@ def __getitem__(self, key: int) -> 'cirq.Moment':
 
     def __getitem__(self, key):
         if isinstance(key, slice):
-            return Circuit(self._moments[key], device=self.device)
+            sliced_circuit = Circuit(device=self.device)
```

Comment:
in this case does it actually cause a speed difference, since this only appends a single moment?

Reply:
The key is a slice.  It's not just one element.  So, if you do large_circuit[2:], it copies the whole circuit to do the slice.  If the circuit is big, doing this slice can be a surprisingly inefficient waste of time.

Reply:
ah got it, a signature on the key would be helpful